tests/difference/core/quint_model/ccv_test.qnt

```diff
@@ -0,0 +1,255 @@
+                )
+            )
+        )
+
+    pure val DeliverPacketToProviderTimeoutTest_resultAndTimeout = deliverPacketToProvider(DeliverPacketToProviderTimeoutTest_testState, "sender")
+
+    run DidTimeOut_DeliverPacketToProviderTimeoutTest =
+    {
+        val result = DeliverPacketToProviderTimeoutTest_resultAndTimeout._1
+        val timedOut = DeliverPacketToProviderTimeoutTest_resultAndTimeout._2
+        val newProviderState = result.newState.providerState
+        val newConsumerState = result.newState.consumerStates.get("sender")
+        not(result.hasError) and
+        timedOut
+    }
+
+    run StateModificationOK_DeliverPacketToProviderTimeoutTest =
+    {
+        val result = DeliverPacketToProviderTimeoutTest_resultAndTimeout._1
+        val timedOut = DeliverPacketToProviderTimeoutTest_resultAndTimeout._2
+        val newProviderState = result.newState.providerState
+        val newConsumerState = result.newState.consumerStates.get("sender")
+        not(result.hasError) and
+        newProviderState.receivedMaturations.size() == 0 and
+        newProviderState.consumerStatus.get("sender") == TIMEDOUT
+    }
+
+    run ConsumerStatusMapHappyPathTest = 
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain1"),
+            Set("chain2"),
+            Set()
+        )
+        res._2 == "" and
+        res._1.get("chain1") == RUNNING and
+        res._1.get("chain2") == STOPPED and
+        res._1.get("chain3") == STOPPED
+    }
+
+    run ConsumerStatusMapAlreadyRunningTest =
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain2"),
+            Set("chain3"),
+            Set()
+        )
+        res._2 == "cannot start a consumer that is stopped or already a consumer"
+    }
+
+    run ConsumerStatusMapAlreadyStoppedTest =
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain1"),
+            Set("chain3"),
+            Set()
+        )
+        res._2 == "Cannot stop a consumer that is not running"
+    }
+
+    run ChainBothInStartAndStopTest =
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain1"),
+            Set("chain1"),
+            Set()
```

Comment:
Is the timing out of chains something we would like to test as well?

Reply:
We could, but I think it's not worth it - it shares a lot of code with the stopped option anyways